runtime/jcl/unix/syshelp.c

```diff
@@ -257,60 +257,100 @@ char *getPlatformFileEncoding(JNIEnv * env, char *codepageProp, int propSize, in
 	char *codepage = NULL;
 	int i = 0;
 	int nameIndex = 0;
-#if defined(LINUX)
+#if defined(LINUX) || defined(OSX)
 	IDATA result = 0;
 	char langProp[24] = {0};
+#if defined(LINUX)
 	char *ctype = NULL;
-	PORT_ACCESS_FROM_ENV(env);
 #endif /* defined(LINUX) */
+	PORT_ACCESS_FROM_ENV(env);
+#endif /* defined(LINUX) || defined(OSX) */
 
 	/* Called with codepageProp == NULL to initialize the locale */
 	if (NULL == codepageProp) {
 		return NULL;
 	}
 
-#ifdef LINUX
+#if defined(LINUX)
 	/*[PR 104520] Return EUC_JP when LC_CTYPE is not set, and the LANG environment variable is "ja" */
-    ctype = setlocale(LC_CTYPE, NULL);
-    if (!ctype || !strcmp(ctype, "C") || !strcmp(ctype, "POSIX")) {
-        result = j9sysinfo_get_env("LANG", langProp, sizeof(langProp));
-        if (!result && !strcmp(langProp, "ja")) {
-            return "EUC-JP-LINUX";
-        }
-    }
+	ctype = setlocale(LC_CTYPE, NULL);
+	if ((NULL == ctype) 
+		|| (0 == strcmp(ctype, "C"))
+		|| (0 == strcmp(ctype, "POSIX"))
+	) {
+		result = j9sysinfo_get_env("LANG", langProp, sizeof(langProp));
+		if ((0 == result) && (0 == strcmp(langProp, "ja"))) {
+			return "EUC-JP-LINUX";
+		}
+	}
 	codepage = nl_langinfo(_NL_CTYPE_CODESET_NAME);
 #elif defined(ARM_EMULATED)	|| defined(CHORUS)
 	codepage = NULL;
+#elif defined(OSX)
+	/* LC_ALL overwrites LC_CTYPE or LANG;
+	 * LC_CTYPE applies to classification and conversion of characters, and to multibyte and wide characters;
+	 * LANG specifies the locale to use except as overridden by LC_CTYPE
+	 */
+	result = j9sysinfo_get_env("LC_ALL", langProp, sizeof(langProp));
```

Comment:
Can we set `codepage = "UTF-8";` and remove all the else clauses?

Reply:
There are a few scenarios such as `LC_ALL`, `LC_CTYPE` or `LANG` is set to `C` and not overwritten by other settings, the code page is expected to be `US-ASCII`. Same test suite `java/lang/System/MacEncoding/MacJNUEncoding.java.MacJNUEncoding` does such test as well.

Reply:
Aha, got it now. Fixed, please review again.